### PR TITLE
Fix/email failed permission check

### PR DESCRIPTION
**Pull request by PedroErnst**
Improved error handling and logging for errors when sending emails. 

## Description

1. No longer logs the user out when sending and email if the  userIsAllowedToSendEmail() function returns false. It is still logged logged with security level. 
This was very annoying when it happens to a user which is not actually trying anything malicious (getting logged out without knowing why), and wouldn't actually stop a malicious actor since they can log in again straight away.

3. Adds more complex logging to the userIsAllowedToSendEmail() function. Currently this function checks whether the user has access to the requested Inbound Account, whether the addresses match and whether he is allowed to use the Outbound Account. But if any of these fails it will make a generic log entry without telling which of the three checks failed. 
So we add a more specific log for each of these cases with extra details.

screenshot from 2018-04-16 14-28-32

## Motivation and Context

Issue #5692 and similar.

## How To Test This

Go to compose Email screen.
Select a from address.
In a different tab, go into the admin or profile settings and change some details from the account that is selected in the first tab.
Try to send the email and fail.
Check the logs.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

File: modules/Emails/EmailsController.php [hunk 1]
```diff
@@ -200,14 +200,12 @@ public function action_send()
             );
 
             $this->view = 'ajax';
-            $response['errors'] = array(
+            $response['errors'] = [
```
Comment (samus-aran) [inline modules/Emails/EmailsController.php]: Not backwards compatible for PHP 5.5.9

Comment (PedroErnst) [reply modules/Emails/EmailsController.php]: I think short array syntax was introduced in [5.4](http://php.net/manual/en/language.types.array.php#example-56)

Comment (gymad) [reply modules/Emails/EmailsController.php]: tested in 5.5.9: it's works for me

**Comment by Dillon-Brown** [comment]
Assessed

**Pull request merged**